classes/PaymentModule.php

```diff
@@ -637,8 +636,13 @@ public function validateOrder($id_cart, $id_order_state, $amount_paid, $payment_
                                 );
                             }
 
-                            $values['tax_incl'] = $order->total_products_wt - $total_reduction_value_ti;
-                            $values['tax_excl'] = $order->total_products - $total_reduction_value_tex;
+                            if ($voucher->free_shipping == 1) {
+                                 $values['tax_incl'] = $order->total_products_wt + $order->total_shipping_tax_incl - $total_reduction_value_ti;
+                                 $values['tax_excl'] = $order->total_products + $order->total_shipping_tax_excl - $total_reduction_value_tex;
+                            } else {
+                                $values['tax_incl'] = $order->total_products_wt - $total_reduction_value_ti;
+                                $values['tax_excl'] = $order->total_products - $total_reduction_value_tex;    
+                            }
```

Comment:
In order to increase readability, I suggest replacing code on lines 639-645 by this one :
```
$values['tax_incl'] = $order->total_products_wt - $total_reduction_value_ti;
$values['tax_excl'] = $order->total_products - $total_reduction_value_tex;

if (1 == $voucher->free_shipping) {
    $values['tax_incl'] += $order->total_shipping_tax_incl;
    values['tax_excl'] += $order->total_shipping_tax_excl;
}
```
As a rule or thumb, you should always try to avoid the `else` part.